src/wp-includes/rest-api/endpoints/class-wp-rest-block-navigation-areas-controller.php

```diff
@@ -0,0 +1,285 @@
+		$data = array();
+		foreach ( get_navigation_areas() as $name => $description ) {
+			$area          = $this->get_navigation_area_object( $name );
+			$area          = $this->prepare_item_for_response( $area, $request );
+			$data[ $name ] = $this->prepare_response_for_collection( $area );
+		}
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if a given request has access to read a navigation area.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_view',
+				__( 'Sorry, you are not allowed to view navigation areas.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+		if ( ! array_key_exists( $request['area'], get_navigation_areas() ) ) {
+			return new WP_Error( 'rest_navigation_area_invalid', __( 'Invalid navigation area.', 'gutenberg' ), array( 'status' => 404 ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks if a request has access to update the specified term.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return bool|WP_Error True if the request has access to update the item, false or WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		return $this->get_item_permissions_check( $request );
+	}
+
+	/**
+	 * Retrieves a specific navigation area.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$name = $request['area'];
+		$area = $this->get_navigation_area_object( $name );
+		$data = $this->prepare_item_for_response( $area, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Updates a specific navigation area.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$name = $request['area'];
+
+		$mapping          = get_option( 'fse_navigation_areas', array() );
+		$mapping[ $name ] = $request['navigation'];
+		update_option( 'fse_navigation_areas', $mapping );
+
+		$area = $this->get_navigation_area_object( $name );
+		$data = $this->prepare_item_for_response( $area, $request );
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Converts navigation area name to a convenient object that this endpoint can reason about.
+	 *
+	 * @param string $name Navigation area name.
+	 * @return stdClass An object representation of the navigation area.
+	 */
+	private function get_navigation_area_object( $name ) {
```

Comment:
```suggestion
	protected function get_navigation_area_object( $name ) {
```

We should give extending classes the flexibility here I think.